### PR TITLE
docs(config): total rework 

### PR DESCRIPTION
**Pull request by adnanrahic**
Fix agent config, fix config root page, add connecting to tracing backends overview page.

#3754

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Loom video

Add your loom video here if your work can be visualized

### CODE REVIEW

**Review by jorgeepc — APPROVED** [review]
🔥

File: docs/docs/configuration/connecting-to-data-stores/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- [Grafana Tempo](/configuration/connecting-to-data-stores/tempo)
+- [Honeycomb](/configuration/connecting-to-data-stores/honeycomb)
+- [Jaeger](/configuration/connecting-to-data-stores/jaeger)
+- [Lightstep](/configuration/connecting-to-data-stores/lightstep)
+- [New Relic](/configuration/connecting-to-data-stores/new-relic)
+- [OpenSearch](/configuration/connecting-to-data-stores/opensearch)
+- [OpenTelemetry Collector](/configuration/connecting-to-data-stores/opentelemetry-collector)
+- [SignalFX](/configuration/connecting-to-data-stores/signalfx)
+- [SigNoz](/configuration/connecting-to-data-stores/signoz)
+- [Sumo Logic](/configuration/connecting-to-data-stores/sumologic)
```
Comment (danielbdias) [inline docs/docs/configuration/connecting-to-data-stores/overview.mdx:42]: There is one missing backend here: Instana
We have a page for it, but it could be good to review it in another PR.

```suggestion
- [AWS X-Ray](/configuration/connecting-to-data-stores/awsxray)
- [Azure App Insights](/configuration/connecting-to-data-stores/azure-app-insights)
- [Datadog](/configuration/connecting-to-data-stores/datadog)
- [Dynatrace](/configuration/connecting-to-data-stores/dynatrace)
- [Elastic APM](/configuration/connecting-to-data-stores/elasticapm)
- [Grafana Tempo](/configuration/connecting-to-data-stores/tempo)
- [Honeycomb](/configuration/connecting-to-data-stores/honeycomb)
- [Instana](/configuration/connecting-to-data-stores/instana)
- [Jaeger](/configuration/connecting-to-data-stores/jaeger)
- [Lightstep](/configuration/connecting-to-data-stores/lightstep)
- [New Relic](/configuration/connecting-to-data-stores/new-relic)
- [OpenSearch](/configuration/connecting-to-data-stores/opensearch)
- [OpenTelemetry Collector](/configuration/connecting-to-data-stores/opentelemetry-collector)
- [SignalFX](/configuration/connecting-to-data-stores/signalfx)
- [SigNoz](/configuration/connecting-to-data-stores/signoz)
- [Sumo Logic](/configuration/connecting-to-data-stores/sumologic)
```

**Review by danielbdias — APPROVED** [review]
